Include/cpython/longintrepr.h

```diff
@@ -139,6 +141,46 @@ _PyLong_CompactValue(PyLongObject *op)
 #define PyUnstable_Long_CompactValue _PyLong_CompactValue
 
 
+/* --- Import/Export API -------------------------------------------------- */
+
+typedef struct PyUnstable_LongLayout {
+    // Bits per digit
+    uint8_t bits_per_digit;
+
+    // Digit size in bytes: sizeof(digit)
+    uint8_t digit_size;
+
+    // Word endian:
+    // - 1 for most significant byte first (big endian)
+    // - 0 for least significant first (little endian)
+    int8_t word_endian;
+
+    // Array endian:
+    // - 1 for most significant byte first (big endian)
+    // - 0 for least significant first (little endian)
+    int8_t array_endian;
+} PyUnstable_LongLayout;
+
+PyAPI_DATA(const PyUnstable_LongLayout) PyUnstable_Long_LAYOUT;
+
+PyAPI_FUNC(PyObject*) PyUnstable_Long_Import(
+    int negative,
+    size_t ndigits,
+    Py_digit *digits);
+
+typedef struct PyUnstable_LongExport {
+    PyLongObject *obj;
+    int negative;
+    size_t ndigits;
+    Py_digit *digits;
+} PyUnstable_LongExport;
+
+PyAPI_FUNC(int) PyUnstable_Long_Export(
+    PyLongObject *obj,
```

Comment:
We should still be using `PyObject *` here, as our creation APIs only create `PyObject *`. Callers cannot currently be expected to know about our internal "subclasses". (And then we should do a real type check in the implementation and raise a Python exception if it's the wrong type.)

Reply:
I'm fine with changing the type to `PyObject*`: done.